server/src/handlers/http/otel.rs

```diff
@@ -0,0 +1,281 @@
+use serde_json::Value;
+mod proto;
+use crate::handlers::http::otel::proto::logs::v1::LogRecordFlags;
+use crate::handlers::http::otel::proto::logs::v1::LogsData;
+use crate::handlers::http::otel::proto::logs::v1::SeverityNumber;
+use std::collections::BTreeMap;
+// Value can be one of types - String, Bool, Int, Double, ArrayValue, AnyValue, KeyValueList, Byte
+fn collect_json_from_any_value(
+    key: &String,
+    value: super::otel::proto::common::v1::Value,
+) -> BTreeMap<String, Value> {
+    let mut value_json: BTreeMap<String, Value> = BTreeMap::new();
+    if value.str_val.is_some() {
+        value_json.insert(
+            key.to_string(),
+            Value::String(value.str_val.as_ref().unwrap().to_owned()),
+        );
+    }
+    if value.bool_val.is_some() {
+        value_json.insert(key.to_string(), Value::Bool(value.bool_val.unwrap()));
+    }
+    if value.int_val.is_some() {
+        value_json.insert(
+            key.to_string(),
+            Value::Number(serde_json::Number::from(value.int_val.unwrap())),
+        );
+    }
+    if value.double_val.is_some() {
+        value_json.insert(
+            key.to_string(),
+            Value::Number(serde_json::Number::from_f64(value.double_val.unwrap()).unwrap()),
+        );
+    }
+
+    //ArrayValue is a vector of AnyValue
+    //traverse by recursively calling the same function
+    if value.array_val.is_some() {
+        let array_val = value.array_val.as_ref().unwrap();
+        let values = &array_val.values;
+
+        for value in values {
+            let value = &value.value;
+            value_json = collect_json_from_any_value(key, value.clone());
+        }
+    }
+
+    //KeyValueList is a vector of KeyValue
+    //traverse through each element in the vector
+    if value.kv_list_val.is_some() {
+        let kv_list_val = value.kv_list_val.unwrap();
+        for key_value in kv_list_val.values {
+            let value = key_value.value;
+            value_json = collect_json_from_values(&value, key);
+        }
+    }
+    if value.bytes_val.is_some() {
+        value_json.insert(
+            key.to_string(),
+            Value::String(value.bytes_val.as_ref().unwrap().to_owned()),
+        );
+    }
+
+    value_json
+}
+
+//traverse through Value by calling function ollect_json_from_any_value
+fn collect_json_from_values(
+    values: &Option<super::otel::proto::common::v1::Value>,
+    key: &String,
+) -> BTreeMap<String, Value> {
+    let mut value_json: BTreeMap<String, Value> = BTreeMap::new();
+
+    for value in values.iter() {
+        value_json = collect_json_from_any_value(key, value.clone());
+    }
+
+    value_json
+}
+
+pub fn flatten_otel_logs(body: &Bytes) -> Vec<BTreeMap<String, Value>> {
+    let mut vec_otel_json: Vec<BTreeMap<String, Value>> = Vec::new();
+    let body_str = std::str::from_utf8(body).unwrap();
+
+    let message: LogsData = serde_json::from_str(body_str).unwrap();
+    for records in message.resource_logs.iter() {
```

Comment:
calling `.iter()` on Option types makes no sense.  There are too many loops that are not needed.

We can just use [opentelemetry-proto](https://crates.io/crates/opentelemetry-proto)
we wont have to deal with the proto file to rust structs generation. And as a added bonus it will get much less complicated.